File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battler/BattlerFactory.scala
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battler

import java.time.LocalDateTime

import org.springframework.social.twitter.api.TwitterProfile

object BattlerFactory extends BattlerParameterGenerator {

  /** s
    * Twitterのプロファイルオブジェクトを元にBattlerオブジェクトを生成する。
    *
    * @param profile Twitterからのプロフィールオブジェクト。
    * @return ゲーム内のBattlerオブジェクト。
    */
  def create(profile: TwitterProfile): Battler = create(profile, LocalDateTime.now())

  /**
    * Twitterのプロファイルオブジェクトを元にBattlerオブジェクトを生成する。
    *
    * @param profile Twitterからのプロフィールオブジェクト。
    * @return ゲーム内のBattlerオブジェクト。
    */
  def create(profile: TwitterProfile, firstSignUpDate: LocalDateTime): Battler = generateBattler(profile, firstSignUpDate)

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battler/BattlerFactory.java
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battler;

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.twitter.TwitterUser;

import java.time.LocalDateTime;

public class BattlerFactory extends BattlerParameterGenerator {
    /**
     * Twitterのプロファイルオブジェクトを元にBattlerオブジェクトを生成する。
     *
     * @param profile Twitterからのプロフィールオブジェクト。
     * @return ゲーム内のBattlerオブジェクト。
     */
    public Battler create(TwitterUser profile) {
        return create(profile, LocalDateTime.now());
    }

    /**
     * Twitterのプロファイルオブジェクトを元にBattlerオブジェクトを生成する。
     *
     * @param profile Twitterからのプロフィールオブジェクト。
     * @return ゲーム内のBattlerオブジェクト。
     */
    public Battler create(TwitterUser profile, LocalDateTime firstSignUpDate) {
        return generateBattler(profile, firstSignUpDate);
    }
}
